src/index.ts

```diff
@@ -1,102 +1,225 @@
+    const username = core.getInput("username", { required: true });
+    const password = core.getInput("password", { required: true });
+    const tlsVerify = core.getInput("tls-verify");
+    const digestFileInput = core.getInput("digestfile");
+
+    // get Podman cli path
+    const podmanPath = await io.which("podman", true);
+
+    let imageToPush = `${imageInput}:${tag}`;
+
+    // check if image exist in Podman storage
+    const isPresentInPodman: boolean = await checkImageInPodman(
+        imageToPush,
+        podmanPath,
+    );
+
+    // check if image exist in Docker storage and if exist pull the image to Podman
+    const isPresentInDocker: boolean = await pullImageFromDocker(
+        imageToPush,
+        podmanPath,
+    );
+
+    // failing if image is not found in Docker as well as Podman
+    if (!isPresentInDocker && !isPresentInPodman) {
+        core.setFailed(`Image ${imageToPush} not found in local storage's of Podman or Docker`);
+        return;
     }
-    core.info(pushMsg);
 
-    //check if images exist in podman's local storage
-    const checkImages = await execute(podman, ['images', '--format', 'json']);
+    // boolean value to check if pushed image is from Docker storage
+    let isPushingDockerImage = false;
 
-    const parsedCheckImages = JSON.parse(checkImages.stdout);
+    if (isPresentInPodman && isPresentInDocker) {
+        let isPodmanImageLatest = false;
+        isPodmanImageLatest = await isPodmanLocalImageLatest(
+            imageToPush,
+            podmanPath,
+        );
 
-    // this is to temporarily solve an issue with the case-sensitive of the property field name. i.e it is Names or names??
-    const nameKeyMixedCase = parsedCheckImages[0] && Object.keys(parsedCheckImages[0]).find(key => 'names' === key.toLowerCase());
-    const imagesFound = parsedCheckImages.
-                            filter((image: string) => image[nameKeyMixedCase] && image[nameKeyMixedCase].find((name: string) => name.includes(`${imageToPush}`))).
-                            map((image: string ) => image[nameKeyMixedCase]);
+        if (!isPodmanImageLatest) {
+            core.warning(`The version of ${imageToPush} in the Docker storage is more recent than the version in the Podman storage. The image from the Docker storage will be pushed.`);
+            imageToPush = `${dockerBaseUrl}/${imageToPush}`;
+            isPushingDockerImage = true;
+        }
+        else {
+            core.warning(`The version of ${imageToPush} in the Podman storage is more recent than the version in the Docker storage. The image from the Podman storage will be pushed.`);
+        }
+    }
+    else if (isPresentInDocker) {
+        imageToPush = `${dockerBaseUrl}/${imageToPush}`;
+        core.info(`Image ${imageToPush} was found in the Docker daemon storage, but not in the Podman storage. The image will be pulled into Podman storage, pushed, and then removed from the Podman storage.`);
```

Comment:
I don't know if "docker storage" or "docker daemon storage" is the right term, but we should at least be consistent.

Reply:
@Divyansh42 

for clarity, let's call them "Docker Image storage" and "Podman Image storage".